packages/pwa-kit-runtime/src/utils/logger.js

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Array defining the order of log levels from lowest to highest severity.
+ * The order of log levels is important. It determines the logging threshold.
+ * @type {string[]}
+ */
+const LOG_LEVELS = ['debug', 'info', 'warn', 'error']
+
+// Default log level for the logging threshold. It can be any of the LOG_LEVELS values.
+const DEFAULT_LOG_LEVEL = 'info'
+
+// Default log format to use if not explicitly provided. It can be 'JSON' or 'TEXT'.
+const DEFAULT_LOG_FORMAT = 'TEXT'
+
+const isServerSide = typeof window === 'undefined'
+
+const LOG_LEVEL = isServerSide
+    ? process.env.PWAKIT_LOG_LEVEL || DEFAULT_LOG_LEVEL
+    : DEFAULT_LOG_LEVEL
```

Comment:
🤔 because of `isServerSide`, the log level can be different on server vs client side. 

If I'm trying to debug an issue on the client side with `logger.debug`, then I won't be able to see the log on the client side.

Reply:
I see. I extended the createLogger to accept specific clientConfig optionally.

```
const createLogger = (packageName, clientConfig = {})
```